mw-webapp/src/logic/waysTable/MentoringWaysTable.tsx

```diff
@@ -1,15 +1,54 @@
 import {useEffect, useState} from "react";
+import {Button} from "src/component/button/Button";
+import {Modal} from "src/component/modal/Modal";
+import {UserPreviewDAL} from "src/dataAccessLogic/UserPreviewDAL";
 import {WayPreviewDAL} from "src/dataAccessLogic/WayPreviewDAL";
-import {PropsWithUuid} from "src/logic/waysTable/OwnWaysTable";
-import {waysColumns} from "src/logic/waysTable/waysColumns";
+import {columnHelper, waysColumns} from "src/logic/waysTable/waysColumns";
 import {WaysTable} from "src/logic/waysTable/WaysTable";
+import {UserPreview} from "src/model/businessModelPreview/UserPreview";
 import {WayPreview} from "src/model/businessModelPreview/WayPreview";
 
+/**
+ * Removes mentor from Way and returns new User Preview
+ */
+const removeMentorFromWay = async (userUuid: string, wayPreview: WayPreview) => {
+  const userPreview = await UserPreviewDAL.getUserPreview(userUuid);
+  const mentoringWays = userPreview.mentoringWays.filter((wayUuid) => wayUuid !== wayPreview.uuid);
+  const newUserPreview = new UserPreview({...userPreview, mentoringWays});
+
+  const mentors = wayPreview.mentors.filter((mentor) => mentor.uuid !== userUuid);
+  const newWayPreview = new WayPreview({...wayPreview, mentors});
+
+  await Promise.all([
+    UserPreviewDAL.updateUserPreview(newUserPreview),
+    WayPreviewDAL.updateWayPreview(newWayPreview),
+  ]);
+
+  return newUserPreview;
+};
+
+/**
+ * Mentoring Ways table props
+ */
+interface MentoringWaysTableProps {
+
+  /**
+   * User uuid
+   */
+  uuid: string;
+
+  /**
+   * Function to change user preview state
+   */
+  setUserPreview: React.Dispatch<React.SetStateAction<UserPreview | null>>;
```

Comment:
We agreed to avoid set null in interfaces if it is semantIcally impossible

Reply:
Fixed